ouroboros-network/src/Ouroboros/Network/PeerSelection/RootPeersDNS/DNS/LedgerPeers.hs

```diff
@@ -202,80 +218,97 @@ long_PEER_LIST_LIFE_TIME = 1847 -- a prime number!
 
 -- | Run the LedgerPeers worker thread.
 --
-ledgerPeersThread :: forall m peerAddr.
+ledgerPeersThread :: forall m peerAddr resolver exception.
                      ( MonadAsync m
                      , MonadMonotonicTime m
+                     , MonadThrow m
+                     , Exception exception
                      , Ord peerAddr
                      )
                   => StdGen
                   -> (IP.IP -> Socket.PortNumber -> peerAddr)
                   -> Tracer m TraceLedgerPeers
                   -> STM m UseLedgerAfter
                   -> LedgerPeersConsensusInterface m
-                  -> ([DomainAccessPoint] -> m (Map DomainAccessPoint (Set peerAddr)))
+                  -> DNSActions resolver exception m
                   -> STM m NumberOfPeers
                   -- ^ a blocking action which receives next request for more
                   -- ledger peers
                   -> (Maybe (Set peerAddr, DiffTime) -> STM m ())
                   -> m Void
-ledgerPeersThread inRng toPeerAddr tracer readUseLedgerAfter LedgerPeersConsensusInterface{..} doResolve
-                  getReq putRsp =
-    go inRng (Time 0) Map.empty
+ledgerPeersThread inRng toPeerAddr tracer readUseLedgerAfter
+                  LedgerPeersConsensusInterface{..} dnsActions
+                  getReq putRsp = do
+    dnsSemaphore <- newLedgerAndPublicRootDNSSemaphore
```

Comment:
Is this semaphore shared with root peers as well? I think not because we're not calling `publicRootPeersProvider` from inside `ledgerPeersThread`.

Reply:
Oh, we don't need to share it, because we use either root peers or ledger peers.